xla/backends/gpu/runtime/command_buffer_cmd.h

```diff
@@ -421,6 +425,10 @@ class CommandBufferCmdExecutor {
 
   // Buffer allocations indices referenced by commands in this sequence.
   absl::flat_hash_set<BufferAllocation::Index> allocs_indices_;
+
+  // Map from buffer allocation index to command ids that references it.
+  absl::flat_hash_map<BufferAllocation::Index, absl::flat_hash_set<CommandId>>
```

Comment:
can you instead keep a map:

`absl::flat_hash_map<CommandId, absl::flat_hash_set<BufferAllocation::Index>>`

and use to from command sequence to skip commands that reference buffer allocations that didn't change?

Reply:
Fixed